### PR TITLE
Cooldown api

### PR DESCRIPTION
**Pull request by SplotyCode**
- Allow servers to change IndicatorType at runtime for specific players (e.g cooldown and no cooldown matches in the same server)
- Allow to implement custom Indicator visualizations

### CODE REVIEW

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	public void hide() {
+		sendActionBar("§r");
+	}
+
+	private void sendActionBar(String bar) {
+		PacketWrapper actionBarPacket = PacketWrapper.create(ClientboundPackets1_8.CHAT_MESSAGE, user);
+		actionBarPacket.write(Type.COMPONENT, new JsonPrimitive(bar));
+		actionBarPacket.write(Type.BYTE, (byte) 2);
+
+		PacketUtil.sendPacket(actionBarPacket, Protocol1_8To1_9.class);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java:33]: Don't use the PacketUtil class, use the intended API instead

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	@Override
+	public void hide() {
+		sendActionBar("§r");
+	}
+
+	private void sendActionBar(String bar) {
+		PacketWrapper actionBarPacket = PacketWrapper.create(ClientboundPackets1_8.CHAT_MESSAGE, user);
+		actionBarPacket.write(Type.COMPONENT, new JsonPrimitive(bar));
+		actionBarPacket.write(Type.BYTE, (byte) 2);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java:31]: Add comment to explain the value 2

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			wrapper.write(Type.FLOAT, (float) progress);
+			wrapper.write(Type.VAR_INT, 0);
+			wrapper.write(Type.VAR_INT, 0);
+			wrapper.write(Type.UNSIGNED_BYTE, (short) 0);
+		} else {
+			wrapper.write(Type.UUID, bossUUID);
+			wrapper.write(Type.VAR_INT, 2);
+			wrapper.write(Type.FLOAT, (float) progress);
+		}
+		PacketUtil.sendPacket(wrapper, Protocol1_8To1_9.class, false, true);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java:37]: Don't use the PacketUtil class, use the intended API instead

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	@Override
+	public void hide() {
+		if (bossUUID == null) return;
+		PacketWrapper wrapper = PacketWrapper.create(ClientboundPackets1_9.BOSSBAR, null, user);
+		wrapper.write(Type.UUID, bossUUID);
+		wrapper.write(Type.VAR_INT, 1);
+		PacketUtil.sendPacket(wrapper, Protocol1_8To1_9.class, false, true);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java:46]: Don't use the PacketUtil class, use the intended API instead

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/TitleCooldownVisualization.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		timePacket.write(Type.INT, stay);
+		timePacket.write(Type.INT, fadeOut);
+		PacketWrapper titlePacket = PacketWrapper.create(ClientboundPackets1_8.TITLE, user);
+		titlePacket.write(Type.VAR_INT, 0);
+		titlePacket.write(Type.COMPONENT, new JsonPrimitive(title));
+		PacketWrapper subtitlePacket = PacketWrapper.create(ClientboundPackets1_8.TITLE, user);
+		subtitlePacket.write(Type.VAR_INT, 1);
+		subtitlePacket.write(Type.COMPONENT, new JsonPrimitive(subTitle));
+
+		PacketUtil.sendPacket(titlePacket, Protocol1_8To1_9.class);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/TitleCooldownVisualization.java:44]: Don't use the PacketUtil class, use the intended API instead

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	public BossBarVisualization(UserConnection user) {
+		this.user = user;
+	}
+
+	@Override
+	public void show(double progress) {
+		PacketWrapper wrapper = PacketWrapper.create(ClientboundPackets1_9.BOSSBAR, user);
+		if (bossUUID == null) {
+			bossUUID = UUID.randomUUID();
+			wrapper.write(Type.UUID, bossUUID);
```
Comment (FlorianMichael) [inline common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/BossBarVisualization.java:25]: Please add comments to all packets you're sending to explain it's content, e. g.

wrapper.write(Type.VAR_INT, 0); // entity id

File: common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java (hunk 1 above)
Comment (SplotyCode) [reply common/src/main/java/com/viaversion/viarewind/protocol/protocol1_8to1_9/cooldown/ActionBarVisualization.java]: should we deprecate PacketUtil then?

**Review by FlorianMichael — APPROVED** [review]
Thanks

**Pull request merged**